fmriprep/workflows/fieldmap/unwarp.py

```diff
@@ -215,10 +221,270 @@ def init_sdc_unwarp_wf(reportlets_dir, ants_nthreads, fmap_bspline,
+                                                      float=True,
+                                                      interpolation='LanczosWindowedSinc'),
+                               name='unwarp_reference')
+
+    ref_msk_post = pe.Node(BETRPT(generate_report=True, frac=0.55),
+                           name='ref_msk_post')
+
+    workflow.connect([
+        (inputnode, explicit_mask_epi, [('in_reference', 'in_file'),
+                                        ('in_mask', 'mask_file')]),
+        (inputnode, cphdr_warp, [('in_reference', 'hdr_file')]),
+        (qwarp, cphdr_warp, [('source_warp', 'in_file')]),
+        (cphdr_warp, to_ants, [('out_file', 'in_file')]),
+        (to_ants, unwarp_reference, [('out', 'transforms')]),
+        (inputnode, unwarp_reference, [('in_reference', 'reference_image')]),
+        (inputnode, unwarp_reference, [('in_reference', 'input_image')]),
+        (unwarp_reference, ref_msk_post, [('output_image', 'in_file')]),
+        (unwarp_reference, outputnode, [('output_image', 'out_reference')]),
+        (ref_msk_post, outputnode, [('mask_file', 'out_mask')]),
+        (ref_msk_post, outputnode, [('out_report', 'out_mask_report')]),
+        (to_ants, outputnode, [('out', 'out_warp')]),
+    ])
+
+    return workflow
+
+
+def init_prepare_epi_wf(ants_nthreads, name="prepare_epi_wf"):
+    """
+    This workflow takes in a set of EPI files with with the same phase 
+    encoding direction and returns a single 3D volume ready to be used in
+    field distortion estimation.
+
+    The procedure involves: estimating a robust template using FreeSurfer's
+    'mri_robust_template', bias field correction using ANTs N4BiasFieldCorrection,
+    and skullstripping using FSL BET.
+
+    .. workflow ::
+
+        from fmriprep.workflows.fieldmap.unwarp import init_prepare_epi_wf
+        wf = init_prepare_epi_wf(ants_nthreads=8)
+
+
+    Inputs
+
+        fmaps
+            list of 3D or 4D NIfTI images
+        ref_brain
+            coregistration reference (skullstripped and bias field corrected)
+
+    Outputs
+
+        out_file
+            single 3D NIfTI file
+
+    """
+    inputnode = pe.Node(niu.IdentityInterface(fields=['fmaps', 'ref_brain']),
+                        name='inputnode')
+
+    outputnode = pe.Node(niu.IdentityInterface(fields=['out_file']),
+                         name='outputnode')
+
+    split = pe.MapNode(fsl.Split(dimension='t'), iterfield='in_file',
+                       name='split')
+
+    merge = pe.Node(
+        StructuralReference(auto_detect_sensitivity=True,
+                            initial_timepoint=1,
+                            fixed_timepoint=True,  # Align to first image
+                            intensity_scaling=True,
+                            # 7-DOF (rigid + intensity)
+                            no_iteration=True,
+                            subsample_threshold=200),
+        name='merge')
```

Comment:
Regarding this: `mri_robust_template` takes either 1 4D image or a series of 3D images.

https://github.com/freesurfer/freesurfer/blob/5367eec84621b331271e97f4e1f1502c3f6d0d87/mri_robust_register/MultiRegistration.cpp#L143-L298